Client/trunk/ParaEngineClient/BlockEngine/RenderableChunk.cpp

```diff
@@ -629,6 +668,46 @@ namespace ParaEngine
 				uint32 nFaceCount = blockmodel.GetFaceCount();
 				instanceGroups[cachedGroupIdx]->AddInstance(i, nFaceCount);
 				totalFaceCount += nFaceCount;
+				// 按照材质分类
```

Comment:
add a if(hasChunkHasMaterial) here 
to avoid iterating all faces for chunks without external materials. (early exit)